src/main/java/subway/Line/application/LineService.java

```diff
@@ -0,0 +1,115 @@
+package subway.Line.application;
+
+import static subway.global.exception.ExceptionCode.NOT_FOUND_LINE;
+import static subway.global.exception.ExceptionCode.NOT_FOUND_STATION;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.Line.domain.Line;
+import subway.Line.domain.Section;
+import subway.Line.domain.Sections;
+import subway.Line.presentation.dto.LineRequest;
+import subway.Line.presentation.dto.LineResponse;
+import subway.Line.infrastructure.LineRepository;
+import subway.Line.presentation.dto.SectionRequest;
+import subway.Line.presentation.dto.SectionResponse;
+import subway.Line.presentation.dto.SectionsResponse;
+import subway.Station.domain.Station;
+import subway.Station.infrastructure.StationRepository;
+import subway.Station.presentation.dto.StationResponse;
+import subway.global.exception.BadRequestException;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Line line = lineRepository.save(toEntity(lineRequest));
+        line.addSection(lineRequest.getUpStationId(), lineRequest.getDownStationId(), lineRequest.getDistance());
+        return createLineResponse(line);
+    }
+
+    private LineResponse createLineResponse(Line line) {
+        return new LineResponse(line.getId(), line.getName(), line.getColor(), line.getDistance(), createSectionsResponse(line.getSections()));
+    }
+
+    private Line toEntity(LineRequest lineRequest) {
+        return new Line(
+                lineRequest.getName(),
+                lineRequest.getColor());
+    }
+
+    public List<LineResponse> findAllLines() {
+        List<Line> lines = lineRepository.findAll();
+
+        return lines.stream()
+                .map(line -> new LineResponse(line.getId(), line.getName(), line.getColor(), line.getDistance(), createSectionsResponse(line.getSections())))
```

Comment:
위에서 언급한 빌더패턴 혹은 정적 팰토리 함수를 이용하면 다음과 같이 작성될 수 있을 것 같네요. 
```suggestion
                .map(LineResponse::from)
```